src/index.js

```diff
@@ -24,46 +26,77 @@ module.exports = class Block {
       throw new Error('second argument must be a CID')
     }
 
-    this._data = data
-    this._cid = cid
+    this.data = data
+    this.cid = cid
+
+    Object.defineProperties(this, {
+      data: readonly,
+      cid: readonly
+    })
   }
 
   /**
    * The data of this block.
    *
+   * @deprecated
    * @type {Uint8Array}
    */
-  get data () {
-    return this._data
-  }
-
-  set data (val) {
-    throw new Error('Tried to change an immutable block')
+  get _data () {
+    deprecateData()
+    return this.data
   }
 
   /**
    * The cid of the data this block represents.
    *
+   * @deprecated
    * @type {CID}
    */
-  get cid () {
-    return this._cid
+  get _cid () {
+    deprecateCID()
+    return this.cid
   }
 
-  set cid (val) {
-    throw new Error('Tried to change an immutable block')
+  get [Symbol.toStringTag] () {
+    return 'Block'
+  }
+
+  get [blockSymbol] () {
+    return true
   }
 
-  // eslint-disable-next-line valid-jsdoc
   /**
    * Check if the given value is a Block.
+   *
+   * @param {any} other
    * @returns {other is Block}
    */
-  static isBlock (other) { // eslint-disable-line no-unused-vars
-    // implemented by class-is module
+  static isBlock (other) {
+    return Boolean(other && other[blockSymbol])
```

Comment:
maybe just `return other && other[blockSymbol] === true`?

Reply:
@rvagg I'm afraid TS will complain because `null && v` is `null` and not a `boolean`.

Reply:
urgh, that's kind of gross, `return !!other && other[blockSymbol] === true`?